src/DataStructures/Mesh.hpp

```diff
@@ -159,6 +159,26 @@ class Mesh {
   template <size_t N = Dim, Requires<(N > 0 and N == Dim)> = nullptr>  // NOLINT
   Mesh<Dim - 1> slice_away(size_t d) const noexcept;
 
+  /*!
+   * \brief Returns a mesh with only the dimensions `d, ...` present.
+   */
+  template <typename... D,
+            Requires<(sizeof...(D) > 0 and sizeof...(D) < Dim)> = nullptr>
```

Comment:
Will thinks it might be useful for generic programming to be able to get the whole Mesh back, that is make the `< Dim` a `<= Dim`

Reply:
test also if this works for getting a 0-dim Mesh, that is `> 0` goes away

Reply:
add appropriate tests when extending the bounds